src/split.c

```diff
@@ -113,9 +113,8 @@ rcutils_split(
 fail:
   error_msg = "unable to allocate memory for string array data";
   if (rcutils_string_array_fini(string_array) != RCUTILS_RET_OK) {
-    error_msg = rcutils_format_string(allocator, "FATAL: %s. Leaking memory", error_msg);
+    RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(allocator, "FATAL: %s. Leaking memory", error_msg);
   }
-  RCUTILS_SET_ERROR_MSG(error_msg, allocator);
```

Comment:
This doesn't look right, when will line 114's message be printed? (the case where finalizing the string array succeeds)

Reply:
I actually keep the original without further checking :), yes, it's not reasonable and gets it tweaked, thanks.